fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/codec/LuceneOptimizedCompoundReader.java

```diff
@@ -71,7 +71,10 @@ public LuceneOptimizedCompoundReader(Directory directory, SegmentInfo si, IOCont
         this.segmentName = si.name;
         String dataFileName = IndexFileNames.segmentFileName(segmentName, "", LuceneOptimizedCompoundFormat.DATA_EXTENSION);
         String entriesFileName = IndexFileNames.segmentFileName(segmentName, "", LuceneOptimizedCompoundFormat.ENTRIES_EXTENSION);
-        handle = directory.openInput(dataFileName, context); // async
+        if ( !(directory instanceof LuceneOptimizedWrappedDirectory) ) {
```

Comment:
Can we change the constructor to take a parameter of the right type and ensure we detect this failure at compile time?